integration/go.mod

```diff
@@ -0,0 +1,178 @@
+module github.com/pulumi/pulumi-cdk/integration
```

Comment:
@blampe was helping us reduce the number of go.mod in the company. Can this just be one top-level go.mod? We are only using Go for a test harness in this repository correct? There's no concern on sharing one between tests and examples?